### PR TITLE
v6.10

### PR DESCRIPTION
**Pull request by Fourdee**
**v6.10**
(03/07/18)

**Changes / Improvements / Optimizations:**

General | ASUS TB: GLES GPU and VPU support now enabled, once Xserver is installed.

General | 'firmware-iwlwifi': Is now a pre-req to WiFi enable. Adds support for Intel based WiFi chips by default: https://github.com/Fourdee/DietPi/issues/1855

General | "net-tools" commands (ifconfig, netstat, route, ...) were replaced by modern "ip" commands (ip a, ip r, ...) within DietPi scripts and the package therefore removed from DietPi core packages: https://github.com/Fourdee/DietPi/issues/1666

General | Removed unused "/DietPi/config.txt" from non-RPi devices: https://github.com/Fourdee/DietPi/pull/1863

General | CurlFTPFS: Removed from DietPi scripts and is no longer supported. Due to lack of security, and, single digit install count (survey).

General | Timesync: DietPi will now only check for a sucessful sync once per system boot, and, again hourly/daily if set. This is to prevent excess delay of systemd-timesyncd service, once the time has already been synced.

General | Sparky SBC: Designs patch added for DSD on MPD-5 dac , new Ids added Mytek Manhatten , LH labs 1V5 2V0 ,HD-AVP/AVA IDA-8: https://github.com/sparky-sbc/sparky-test/tree/master/dsd-marantz

DietPi-Backup | Rewritten. Userdata option removed, included by default backup. Added options to edit include/exclude filters in the GUI. Existing backups (v6.9 or less) are no longer supported: https://github.com/Fourdee/DietPi/issues/1851

DietPi-Config | Soundcards (RPi): Allo Katana, now available for selection. https://github.com/Fourdee/DietPi/issues/1849

DietPi-Config | IntelGPU Driver: Installation code added: https://github.com/Fourdee/DietPi/issues/1855#issue-333150262

DietPi-Config | Networking: You can now view the sent and recieved totals for both network devices. NB: 32bit devices will reset the values after 32bit int limit is reached (roughly 4.3GB~), this is a kernel/arch limitation: https://github.com/Fourdee/DietPi/issues/1666#issuecomment-401546728

DietPi-Drive_Manager | Rewrite and improvements:
 - Now supports ROM devices (eg: DVD). NB: compatibility for DVD/CD devices relies on kernel support. Not all devices will support DVD/CD devices, and/or their filesystem format: https://github.com/Fourdee/DietPi/issues/1858
 - Resize ext4 options added: https://github.com/Fourdee/DietPi/issues/1821
 - Support for detecting and formatting non-partitioned drives
 - You can now benchmark read/write for all available mounted drives: https://github.com/Fourdee/DietPi/issues/1858
 - Now supports mounting, viewing, removal of network drives on system (nfs4, cifs). This replaces the previous options in 'dietpi-config': https://github.com/Fourdee/DietPi/issues/1858
 - Moving of DietPi user data to filesystems that do not support permission (FAT), is no longer supported and disabled: https://github.com/Fourdee/DietPi/issues/1846

DietPi-Globals | G_BACKUP: Added to globals. This will be gradually rolled out in DietPi scripts, allowing the user to create a system backup, prior to significant system changes: https://github.com/Fourdee/DietPi/issues/1871#issuecomment-400443401

DietPi-Globals | G_CHECK_USERDATA: Added to DietPi scripts: https://github.com/Fourdee/DietPi/issues/1850#issuecomment-401407996

DietPi-Services | Rsync: Added to service control. This will prevent errors if rsync is running in daemon mode, when 'dietpi-backup' is run: https://github.com/Fourdee/DietPi/issues/1869#issuecomment-399890771

DietPi-Software | Ubooquity: Now available for installation: https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=12969#p12969 (https://github.com/Fourdee/DietPi/issues/1845#issuecomment-397447909)

DietPi-Software | Roon Extension Manager: Now available for installation. Many thanks to @JanKoudijs for contributing this great addition!: https://github.com/Fourdee/DietPi/pull/1865

DietPi-Software | Mosquitto: Now uses Debian APT installation: https://github.com/Fourdee/DietPi/issues/1868#issuecomment-399982278

DietPi-Software | Gogs: Now runs under its own user. For new installations only: https://github.com/Fourdee/DietPi/issues/1877

DietPi-Software | Xserver: 'mesa-utils-extra' now also installed by default, useful for GLES testing 'es2_info es2gears' etc.

DietPi-Software | Moode: Removed from our software lineup: https://github.com/Fourdee/DietPi/issues/1223#issuecomment-401549371

DietPi-Software | Radarr, Sonarr and Jackett: Services optimized and no longer run in debugging mode. Should improve runtime performance. Many thanks to @userdeveloper98 for this contribution! https://github.com/Fourdee/DietPi/pull/1889

DietPi-Survey | Simplified available options. You can now either Opt In, or, Opt Out and automatically have any existing data cleared. Interactive installations will be prompted to Opt In or Out during 1st run. Automated installations are Opted In by default, you can change this once setup is completed. More information on DietPi-Survey and how to change the options: https://dietpi.com/phpbb/viewtopic.php?f=8&t=20 https://github.com/Fourdee/DietPi/issues/1827#issuecomment-396005575

PREP | 'os-prober' installed by default for x86_64 devices. Ensures dual boot OSs are detected by grub. Also added a 3 second timeout to grub boot prompt, allowing OS selection: https://github.com/Fourdee/DietPi/issues/1855

**Bug Fixes:**

ASUS TB | Resolved square (broken) X11 fonts.

Raspberry Pi | Removed "initial_turbo" setting from DietPi-Config and config.txt, as it prevents CPU governor from throttling down: https://github.com/Fourdee/DietPi/issues/1836

DietPi-Drive_Manager | Resolved incorrect detection of available drives: https://github.com/Fourdee/DietPi/issues/1858

DietPi-Software | GMrender: Resolved an issue where two systems on the same network would nullify the other. Hostname is now used for the server name, UUID used is applied via DietPi generated UUID during 1st run: https://dietpi.com/phpbb/viewtopic.php?f=11&t=3900&p=12985#p12985

DietPi-Software | Apache2: Fixed a syntax error that leads to Apache logging to "/error.log" instead of "/var/log/apache2/error.log"

DietPi-Software | Nukkit: Fixed the broken download link on installation. Many thanks to @symbios24 for reporting bug and providing solution: https://github.com/Fourdee/DietPi/issues/1875

DietPi-Software | Linux software: Resolved an issue with NULL entry being displayed: https://github.com/Fourdee/DietPi/pull/1830#issuecomment-401612168

DietPi-Config | Fixen an issue, where IPv6 could not be disabled on RPi. On current kernel version it is no dedicated kernel module any more and needs to be toggled via "/boot/cmdline.txt".

**AlloGUI v9:**
- Changing the root password, no longer breaks web interface: https://github.com/Fourdee/DietPi/issues/1841
- Resolved issues with terminal leakage in the web interface: https://github.com/Fourdee/DietPi/issues/1841
- System settings: Current version is always shown, even if an update is available.
- System settings: Allo Katana, now available for selection. https://github.com/Fourdee/DietPi/issues/1849

### CODE REVIEW

**Comment by Fourdee** [comment]
@MichaIng 

Release candidate for release tomorrow.

Pending from my side:
- Resolve or disable rootFS transfer

**Comment by MichaIng** [comment]
@Fourdee 
- [x] Tiny on DietPi-PREP: Forgot to press `Start  a preview` button 😅, but just a minor anyway. Fixed: https://github.com/Fourdee/DietPi/pull/1881/commits/7e8b25c8072b502ffcff56d92489a2fb6e3259aa
- [x] Minor on G_DIETPI-NOTIFY colorizing (fixed: https://github.com/Fourdee/DietPi/pull/1881/commits/a25b67ad5b4145a263d654225a800c82ca0db170):
colorissue
  - Found the reason: Color codes > 47m are not understood by all terminals, e.g. `\e[90m` is actually bright black (gray), but to allow this within all terminal, we need to use `\e[1;30m`. Same for the animation dot color, where one appeared as blue, because 93 (bright yellow) is not understood, instead `\e[1;33m` perfectly works. Combination of bright and the 9X colors anyway do not change something. So:
colorissue
- [x] Same for DietPi-Banner colourizing: https://github.com/Fourdee/DietPi/pull/1881/commits/7a56d17d86c61b969e38dcba774144efa5f1c971
  - Could need some clean up still, don't know about this `\e[38;5;154m`, looks like RGB code? It seems to be supported by all terminals, but on Jessie VM terminal it is finally just shown as bright yellow, maybe due to limited amount of supported colours.
- [ ] DietPi-Drive_Manager still finds and allows handling of empty ROM drives. Format etc of course fails:
rom
- [ ] Error on formatting totally empty drive (no partitions, no UUID), due to `aDRIVE_MOUNT_TARGET[$INDEX_INIT_CURRENT_DEVICE]=${aDRIVE_MOUNT_SOURCE[$INDEX_INIT_CURRENT_DEVICE]}` and `rm -R $FORMAT_PREVIOUS_MOUNT_TARGET`.
  - You already faced this here: https://github.com/Fourdee/DietPi/issues/1858#issuecomment-400312970
- [ ] DietPi-Survey upgrade, **PR is up, WIP: https://github.com/Fourdee/DietPi/pull/1884**
  - What you thinks about https://github.com/Fourdee/DietPi/issues/1827#issuecomment-397304683 ? For the first it is less flexible, than doing your approach with own array index for every upload file, because the connections between e.g. used device and installed software is lost (could be interesting as well if certain software/setting is mostly used on certain device, or if certain software is mostly installed together with certain other software), but it is way faster, if we are for now just interested in the overall counts. Using `grep` on the files still allows to check for combination of values, but is then due to speed again not useful for automatic report page creation.
  - Secondly: https://github.com/Fourdee/DietPi/issues/1827#issuecomment-399689518. You want to keep survey upload count, or remove it? I will add the automation count.

**Comment by Fourdee** [comment]
@MichaIng 

> ROM

I'll get this implemented properly in v6.11, needs more time than I have available today.

note to self:
```
		#Set flags for drives
		for ((i=0; i<${#aDRIVE_MOUNT_SOURCE[@]}; i++))
		do

			#ROM (eg: cd)
			if [[ $aDRIVE_MOUNT_SOURCE == '/dev/sr'* ]]; then

				aDRIVE_ISROM[$i]=1

			fi

		done

#or in menu
if [[ $aDRIVE_MOUNT_SOURCE == '/dev/sr'* ]]; then
arrr this be a CD rom!
```

> Error on formatting totally empty drive (no partitions, no UUID), due to 

Did you get the reboot prompt? Once rebooted, the drive should be detected correctly? Tried partprobe and various items, but it seems only a reboot reloads basic level drive info.

**Comment by MichaIng** [comment]
@Fourdee 
>Did you get the reboot prompt? Once rebooted, the drive should be detected correctly? Tried partprobe and various items, but it seems only a reboot reloads basic level drive info.

Somehow I don't get the reboot prompt. But as said, this is easily fixable. I didn't touch drive_manager due to your ongoing work on this, but can try to fix it.

What do you think about in case move release for tomorrow? I also want to have this survey file adjustment finished, but wanted to wait for your thoughts about this 😉. I can do that until the night, but until tomorrow we can surely fix the left drive_manager quirks. "Scan-on-reboot" excluded => v6.11

**Comment by Fourdee** [comment]
> What do you think about in case move release for tomorrow?

@MichaIng 

Yep agree 👍 Tomorrow it is.

**Comment by Fourdee** [comment]
REF: https://twitter.com/DietPi_/status/1013438979792875521

**Comment by Fourdee** [comment]
>  I didn't touch drive_manager due to your ongoing work on this, but can try to fix it.

@MichaIng 

Yep, go for it, i'll resolve the conflicts if any 👍

**Comment by Fourdee** [comment]
New visual style:

Note to self:
- Get max char length of whip array (index 1/odd only, ignore 0/even), then set to length of -
image

**Comment by MichaIng** [comment]
Investigating, when I wanted to opt in DietPi-Survey:
```
root@VM-Jessie:~# /DietPi/dietpi/dietpi-survey
[  OK  ] Root access verified.
/DietPi/dietpi/func/dietpi-globals: line 558: ((: 551 0 47 11 27 24 27 26 27 44 33 19 29 46 30 30 27 27 27 43 0 37 > ( 120 - 4 ) : syntax error in expression (error token is "0 47 11 27 24 27 26 27 44 33 19 29 46 30 30 27 27 27 43 0 37 > ( 120 - 4 ) ")
[  OK  ] DietPi-Survey | Connection test: dietpi.com
[  OK  ] DietPi-Survey | Successfully sent survey data
```

@Fourdee 
Double quotes missing to keep line breaks within string:  `done <<< "$character_count_per_line"`
**€: Fixed: https://github.com/Fourdee/DietPi/commit/d7d5e3f5c46c8ea6ecf3f3662307cedeb01e2cef**

As it has UI, make `dietpi-survey` part of dietpi-* aliases, so path doesn't need to be typed in terminal?

File: dietpi/patch_file [hunk 1]
```diff
@@ -725,20 +709,128 @@ _EOF_
...
+			#-------------------------------------------------------------------------------
+			#Remove config.txt for non-RPi devices: https://github.com/Fourdee/DietPi/pull/1863
+			if (( $G_HW_MODEL >= 10 )); then
+
+				[[ -f /DietPi/config.txt ]] && rm /DietPi/config.txt
+				[[ -f /boot/config.txt ]] && rm /boot/config.txt
+
+			fi
+			#-------------------------------------------------------------------------------
+			#Removal of NTP from dietpi-software
```
Comment (MichaIng) [inline dietpi/patch_file:765]: @Fourdee 
curlftpfs (index 2) needs to be reset as well?

**Comment by Fourdee** [comment]
@MichaIng 

I'am nearly done for the day, lets move this back to tomorrow https://twitter.com/DietPi_/status/1013915130801868801

> As it has UI, make dietpi-survey part of dietpi-* aliases, so path doesn't need to be typed in terminal?

Yep, go for it 👍 

> Double quotes missing to keep line breaks within string: done <<< "$character_count_per_line"
€: Fixed: d7d5e3f

Apologies, think I did that eariler

**Comment by MichaIng** [comment]
@Fourdee 
Done + minor GitHub ticket creator tuning: https://github.com/Fourdee/DietPi/commit/6c43c43d04ff406403fb5474aef564275d3407b2

Jep, I think it's best to move it to tomorrow. This drive manager turns out a tricky one 😉. Have a nice sleep 👍.

**Comment by userdeveloper98** [comment]
@Fourdee  @MichaIng 
Hi there guys, I would like to contribute a little bit to your project, could you please grant me write access to the repository ?

--
I've created a branch for Pull Request, but I can't push it :(

**Comment by MichaIng** [comment]
@userdeveloper98 
Go "pull requests" > click "compare across forks" and then you should be able to create pull requests from your own repository to be pushed into Fourdee/testing branch.

Contributions are always welcome of course, but please understand, that @Fourdee will not directly add you as official contributor, I guess, although maybe there is a way to do that with access limited, to edit new branches, just to skip handling an own repo?

**Comment by userdeveloper98** [comment]
@MichaIng 
Thanks for reply, I'll try "forks workaround" ... 👍 

> I guess, although maybe there is a way to do that with access limited, to edit new branches, just to skip handling an own repo?

The branch protection may be enabled and no changes will be possible without Pull Requests, except Admin of sure :)

**Comment by MichaIng** [comment]
@userdeveloper98 
It is not really a workaround but common practice. Should work fine 😀, did this myself for some months.
Short cut is to browse testing repo to the file you want to edit, click edit button will directly create a new branch on you own repo and offers to automatically create PR then to Fourdee/testing branch.

@Fourdee 
https://github.com/Fourdee/DietPi/blob/testing/rootfs/var/lib/dietpi/services/fs_partition_resize.sh#L66
This does only apply for mmc devices but not sdX? Or are the latter covered by one prior command?

**Comment by Fourdee** [comment]
@MichaIng 

> https://github.com/Fourdee/DietPi/blob/testing/rootfs/var/lib/dietpi/services/fs_partition_resize.sh#L66
This does only apply for mmc devices but not sdX? Or are the latter covered by one prior command?

Good spot 👍 Needs fixing. No ```p``` for ```sdX``` based drives.

**Comment by MichaIng** [comment]
@Fourdee 
Hehe, has some benefits so have a casual look into our files. I am currently doing a bid minor coding enhancements, double braces, grep -q, $(<file) instead of $(cat file) and stuff like that.

Found `DietPi-Optimal_MTU`. I don't understand the background or why ping shows the error for large MTUs, but found my optimal MTU of `1492` on google.com and dietpi.com.
Maybe we can implement this into `dietpi-config` by times?

**Comment by Fourdee** [comment]
@MichaIng 

> Maybe we can implement this into dietpi-config by times?

Yep, for v6.11 👍

**Comment by Fourdee** [comment]
@MichaIng 

Appears my opted in SBC's are now opted out during recent patch:
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
5
```

---


Sys1:
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
15
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey^C
root@DietPi:~#  cat /DietPi/dietpi/.dietpi-survey
1
15
```

sys 2:
```
root@DietPi:~# nano /DietPi/dietpi/.dietpi-survey
root@DietPi:~# cat
^C
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
2
5

```

```dietpi-update -1```

sys 1 issue
```
root@DietPi:~# cat /DietPi/dietpi/.dietpi-survey
0
15
```

Should not affect end users, only multiple patches of 6.10, either way, I'll fix.

**Comment by MichaIng** [comment]
@Fourdee 
Jep, when using `dietpi-update -1` this is done. Hmm we could scrape /DietPi/dietpi/.version to get real version. I will add it.

**Comment by Fourdee** [comment]
@MichaIng 

Maybe we should save the survey version in the save file for easy patching?
```
cat << _EOF_ > $fp_temp
$opted_in
$survey_sentcount
$survey_version
_EOF_
```

**Comment by MichaIng** [comment]
@Fourdee 
Also an idea, but I made commit already, which is working fine to keep correct opt status: https://github.com/Fourdee/DietPi/commit/da50ea2f4c0c5e794bdc6fdf6b9d08be594def6a

**Comment by Fourdee** [comment]
@MichaIng 

Thanks 👍 

Ok, few hours of final general testing, then i'll merge if no issues found.

---

Edit, is the new survey code added to server and as you want it?

**Comment by MichaIng** [comment]
@Fourdee 
Jep, doing the same here 🙂.

€:
- `[ INFO ] DietPi-Bugreport | Lines required for Whiptail box and its contents (y=61, z=0), exceeds screen`
  - Remove debug messages. Actually for the future I would like to have `G_DIETPI-NOTIFY 4 message` => `(( $G_DEBUG_MODE ))` && `[ DEBUG] message`. We could leave debug messages at critical spots in place then and also make them available to end user for debugging, G_DIETPI-NOTIFY should just exit fast (check at beginning of function) to no have much delay.
- software index 2 status needs be be reset as well?

**Comment by Fourdee** [comment]
Note to self:
- Move all v6.9 BR's to a archive folder

**Comment by Fourdee** [comment]
@MichaIng 

> Remove debug messages. Actually for the future I would like to have G_DIETPI-NOTIFY 4 message

The main reason for leaving them in, is in a hope, that if the user goes "why doesn't this fit, i'll send a bug report and git ticket", they may see the info, prior to that.

**Comment by Fourdee** [comment]
Ok, lets release.

No major dev changes in the next few days, reserve it for any hotfixes, if they are required. 

@MichaIng 

Great work as always 👍 🥇

**Comment by MichaIng** [comment]
Jep, I agree. Can't wait to see how fast the update is applied according to survey report :smiley:.

**Pull request merged**